internal/templates/commands/render.go

```diff
@@ -0,0 +1,176 @@
+			`"helloworld@v1" means "github.com/abcxyz/helloworld repo at revision v1; this is for a template owned by abcxyz. ` +
+			`"myorg/helloworld@v1" means github.com/myorg/helloworld repo at revision v1; this is for a template not owned by abcxyz but still on GitHub. ` +
+			`"mygithost.com/mygitrepo/helloworld@v1" is for a template in a remote git repo but not owned by abcxyz and not on GitHub. ` +
+			`"mylocaltarball.tgz" is for a template not in git but present on the local filesystem. ` +
+			`"http://example.com/myremotetarball.tgz" os for a non-Git template in a remote tarball.`,
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "spec",
+		Example: "path/to/spec.yaml",
+		Target:  &r.fv.spec,
+		Default: "./spec.yaml",
+		Usage:   "The path of the .yaml file within the unpacked template directory that specifies how the template is rendered.",
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "dest",
+		Aliases: []string{"d"},
+		Example: "/my/git/dir",
+		Target:  &r.fv.dest,
+		Default: ".",
+		Usage:   "Required. The target directory in which to write the output files.",
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "git-protocol",
+		Example: "https",
+		Default: "https",
+		Target:  &r.fv.gitProtocol,
+		Usage:   "Either ssh or https, the protocol for connecting to GitHub. Only used if the template source is GitHub.",
+	})
+	sect.StringMapVar(&cli.StringMapVar{
+		Name:    "input",
+		Example: "foo=bar",
+		Target:  &r.fv.inputs,
+		Usage:   "The key=val pairs of template values; may be repeated.",
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "log-level",
+		Example: "info",
+		Default: "warning",
+		Target:  &r.fv.logLevel,
+		Usage:   "How verbose to log; any of debug|info|warning|error.",
+	})
+	sect.BoolVar(&cli.BoolVar{
+		Name:    "force-overwrite",
+		Target:  &r.fv.forceOverwrite,
+		Default: false,
+		Usage:   "If an output file already exists in the destination, overwrite it instead of failing.",
+	})
+	sect.BoolVar(&cli.BoolVar{
+		Name:    "keep-temp-dirs",
+		Target:  &r.fv.keepTempDirs,
+		Default: false,
+		Usage:   "Preserve the temp directories instead of deleting them normally.",
+	})
+
+	return set
+}
+
+func (r *Render) parseFlags(args []string) error {
+	if err := r.Flags().Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+
+	if r.fv.template == "" {
+		return fmt.Errorf("-template is required")
+	}
+
+	return nil
+}
+
+func (r *Render) Run(ctx context.Context, args []string) error {
+	if err := r.parseFlags(args); err != nil {
+		return err
+	}
+
+	if r.fs == nil { // allow filesystem interaction to be faked for testing
+		r.fs = &realFS{}
+	}
+
+	if err := destOK(r.fs, &r.fv); err != nil {
+		return err
+	}
+
+	return fmt.Errorf("stub")
+}
+
+// destOK makes sure that the output directory looks sane; we don't want to clobber the user's
+// home directory or something.
+func destOK(fs fs, fv *flagValues) error {
```

Comment:
It looks like the only value from `*flagValues` is destination - maybe it would be better to just accept that string as a param? It would be much easier to test too.

Reply:
Done, thanks.